src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -837,6 +839,16 @@ def load_arguments(self, _):
     with self.argument_context('aks trustedaccess rolebinding update') as c:
         c.argument('roles', help='comma-separated roles: Microsoft.Demo/samples/reader,Microsoft.Demo/samples/writer,...')
 
+    with self.argument_context('aks mesh enable-ingress-gateway') as c:
+        c.argument('ingress-gateway-type',
+                   validator=validate_azure_service_mesh_ingress_gateway_type,
```

Comment:
One trick to omit the validator is to leverage the enum type. It will also add the allowed values in help message automatically. One example would be `os_sku`.

Reply:
Thanks.  Updated.